### PR TITLE
[BugFix] : Check if RaumPin is a number. 

### LINKED ISSUE #167: Raumpin ist nummer.  Eingabefeld erlaubt jedoch auch Buchstaben.
Ausschließlich Eingabe von Zahlen erlauben (Regex?) . Sicherheitshalber  Server seitig prüfen ?  

Fehlermeldung bei falscher Pin statt 400 Fehler und redirect.

### PR DESCRIPTION
**Pull request by medkhabt**
closes #167 

Hi,
I'm not a native German, but i hope i got what you meant. 

I made some changes to fix this issue: 
- add a `private void checkRoomPinFormatForRoom(Room room)` to check if the raupin is a number before it saves it. 
- add a unit test to check both cases ( raupin is not a number, raupin is null ).

### CODE REVIEW

**Review by MorMundHS-MA — CHANGES_REQUESTED** [review]
First of all welcome, to the project and thank you very much for your pull request.  
In the future we will try to have all issues in English, but you got most of it right.  
The "RaumPin" is a PIN number you have to enter when checking into a room. It is intended to prevent malicious activity from outside the university, since the website is reachable in the open web.

Your fix looks good so far and thanks for writing tests as well.

However the issue primarily is about the frontend input field on the web page (for example http://localhost:8080/r/test). If you want to you can fix that error as well. Otherwise I'll merge this as is, but leave the issue open.

**Comment by medkhabt** [comment]
Hi there, thank you for explaining the roomPin, it helped me understand the issue. Concerning the front-end input field, i changed its type to 'number', and made a format validation for roomPin in the RoomController where the post request is treated ( by throwing InvalidRoomPinException ). 

I wanted to ask, if it is necessary to make InvalidRoomPinException a checkedException.

File: src/main/java/de/hs_mannheim/informatik/ct/persistence/services/RoomService.java [hunk 1]
```diff
@@ -75,6 +76,17 @@ public Room saveRoom(@NonNull Room room) {
         return roomList;
     }
 
+    private void checkRoomPinFormatForRoom(Room room) {
```
Comment (medkhabt) [inline src/main/java/de/hs_mannheim/informatik/ct/persistence/services/RoomService.java:79]: Should I remove the check i made earlier? i thought at first that the roomPin can be changed by a user, if it is not accessible by him, i think it isn't usefull.

Comment (MorMundHS-MA) [reply src/main/java/de/hs_mannheim/informatik/ct/persistence/services/RoomService.java:79]: No you can keep it. I see no harm in a sanity check.

**Review by MorMundHS-MA — CHANGES_REQUESTED** [review]
> I wanted to ask, if it is necessary to make InvalidRoomPinException a checkedException.

I think in the Java world it should be, yes. 

Thanks for fixing the front end too!
A few small changes are still needed:

- [ ]  You found a bug in `checkInWithOverride`, it didn't check the room PIN at all. Could you integrate the check if the PIN is correct into `roomPinValidation` and fix the incorrect test case in `RoomControllerTest::checkInFullRoomWithOverride`?
- [ ] `type=number` inputs have spin buttons for incrementing/decrementing the value, which is pointless for a PIN. In the student email input we use this CSS to hide them
``` CSS
input.email::-webkit-outer-spin-button,
input.email::-webkit-inner-spin-button {
    display: none;
}

input[type=number].email {
    -moz-appearance: textfield;
}
```

**Comment by medkhabt** [comment]
> > I wanted to ask, if it is necessary to make InvalidRoomPinException a checkedException.
> 
> I think in the Java world it should be, yes.
> 
> Thanks for fixing the front end too!
> A few small changes are still needed:
> 
> * [ ]   You found a bug in `checkInWithOverride`, it didn't check the room PIN at all. Could you integrate the check if the PIN is correct into `roomPinValidation` and fix the incorrect test case in `RoomControllerTest::checkInFullRoomWithOverride`?
> * [ ]  `type=number` inputs have spin buttons for incrementing/decrementing the value, which is pointless for a PIN. In the student email input we use this CSS to hide them
> 
> ```css
> input.email::-webkit-outer-spin-button,
> input.email::-webkit-inner-spin-button {
>     display: none;
> }
> 
> input[type=number].email {
>     -moz-appearance: textfield;
> }
> ```
Hi, i made the requested changes: 

- Made the test `RoomControllerTest::checkInFullRoomWithOverride` pass, by adding a correct pin as you mentioned. 
And made an other test to check the case when the room pin is invalid. 

- Hided the inc/desc icons with the help of your code snippet.

**Review by MorMundHS-MA — APPROVED** [review]
Perfect, again huge thanks for contributing to the project and welcome aboard!

**Comment by medkhabt** [comment]
Thank you for your help, looking forward to contributing more.
